src/Modules/Legacy/Fields/ApplyFilterToFieldData.cc

```diff
@@ -25,88 +25,76 @@
 
-namespace SCIRun {
+ALGORITHM_PARAMETER_DEF(Fields, Erode)
+ALGORITHM_PARAMETER_DEF(Fields, Dilate)
 
-using namespace SCIRun;
-using namespace SCIRunAlgo;
+ApplyFilterToFieldData::ApplyFilterToFieldData() :
+  Module(staticInfo_)
+{
+  INITIALIZE_PORT(InputField);
+  INITIALIZE_PORT(OutputField);
+}
 
 /// @class ApplyFilterToFieldData
 /// @brief Applies a dilate or erode filter to a regular mesh.
 
-class ApplyFilterToFieldData : public Module {
-  public:
-    ApplyFilterToFieldData(GuiContext*);
-    virtual ~ApplyFilterToFieldData() {}
-    virtual void execute();
-
-  private:
-    GuiString method_;
-    GuiString edmethod_;
-    GuiInt edniter_;
-
-    SCIRunAlgo::ErodeFieldDataAlgo  erode_algo_;
-    SCIRunAlgo::DilateFieldDataAlgo dilate_algo_;
-};
-
-
-DECLARE_MAKER(ApplyFilterToFieldData)
-
-ApplyFilterToFieldData::ApplyFilterToFieldData(GuiContext* ctx) :
-  Module("ApplyFilterToFieldData", ctx, Source, "ChangeFieldData", "SCIRun"),
-    method_(ctx->subVar("method")),
-    edmethod_(ctx->subVar("ed-method")),
-    edniter_(ctx->subVar("ed-iterations"))
+void ApplyFilterToFieldData::setStateDefaults()
 {
-  erode_algo_.set_progress_reporter(this);
-  dilate_algo_.set_progress_reporter(this);
+  auto state = get_state();
+  state->setValue(Core::Algorithms::Variables::MaxIterations, 2);
+  state->setValue(Parameters::Erode, true);
+  state->setValue(Parameters::Dilate, false);
 }
 
-
-void
-ApplyFilterToFieldData::execute()
+void ApplyFilterToFieldData::execute()
 {
-  FieldHandle input, output;
+  auto input = getRequiredInput(InputField);
 
-  get_input_handle("Field",input,true);
-
-  if (inputs_changed_ || !oport_cached("Field") || method_.changed() ||
-      edmethod_.changed() || edniter_.changed())
+  if (needToExecute())
   {
-    // Inform module that execution started
-    update_state(Executing);
-
-    if (method_.get() == "erodedilate")
+    auto state = get_state();
+    auto iters = state->getValue(Variables::MaxIterations).toInt();
+    auto doDilate = state->getValue(Parameters::Dilate).toBool();
+    auto doErode = state->getValue(Parameters::Erode).toBool();
+    FieldHandle output;
+    // Dilate then erode
+    if (doDilate)
     {
-      if (edmethod_.get() == "erode")
+      DilateFieldDataAlgo dilate;
+      dilate.set(Variables::MaxIterations, iters);
+      if (!dilate.runImpl(input, output))
       {
-        erode_algo_.set_int("num_iterations",edniter_.get());
-        if(!(erode_algo_.run(input,output))) return;
+        THROW_ALGORITHM_PROCESSING_ERROR("Error in dilate algo");
       }
-      else  if (edmethod_.get() == "dilate")
-      {
-        dilate_algo_.set_int("num_iterations",edniter_.get());
-        if(!(dilate_algo_.run(input,output))) return;
-      }
-      else
+      
+    }
+    if (doErode)
+    {
+      ErodeFieldDataAlgo erode;
+      erode.set(Variables::MaxIterations, iters);
+      if (!erode.runImpl(input, output))
```

Comment:
This implementation will ignore the dilated volume in the case when both both dilate and erode are true

Reply:
Let me fix